src/Raven.Server/Documents/ETL/Providers/Queue/QueueEtl.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CloudNative.CloudEvents;
+using CloudNative.CloudEvents.Extensions;
+using CloudNative.CloudEvents.Kafka;
+using Confluent.Kafka;
+using Raven.Client.Documents.Operations.Counters;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Client.Documents.Operations.ETL.Queue;
+using Raven.Server.Documents.ETL.Metrics;
+using Raven.Server.Documents.ETL.Providers.Queue.Enumerators;
+using Raven.Server.Documents.ETL.Providers.Queue.Test;
+using Raven.Server.Documents.ETL.Stats;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.Exceptions.ETL.QueueEtl;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.ETL.Providers.Queue;
+
+public class QueueEtl : EtlProcess<QueueItem, QueueWithEvents, QueueEtlConfiguration, QueueConnectionString, EtlStatsScope,
+    EtlPerformanceOperation>
+{
+    public QueueEtl(Transformation transformation, QueueEtlConfiguration configuration, DocumentDatabase database, ServerStore serverStore)
+        : base(transformation, configuration, database, serverStore, QueueEtlTag)
+    {
+        Metrics = new EtlMetricsCountersManager();
+    }
+
+    private const string DefaultCloudEventId = "cloud-event-id";
+    private const string DefaultType = "ravendb.etl.put";
+    private const string DefaultSource = "https://ravendb.net/";
```

Comment:
```suggestion
    private const string DefaultSource = "https://ravendb.net/";
```

Need to come up with a better URI, I think.

Reply:
We will changed it to: Node URL + DB name + Task name.